### PR TITLE
#1242 Support tooltip in password control

### PR DESCRIPTION
**Pull request by nabilIbm**
Fixes : #1242  
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/common-canvas/__tests__/common-properties/controls/passwordfield-test.js [hunk 1]
```diff
@@ -175,6 +175,24 @@ describe("Passwordfield renders correctly", () => {
 		const messageWrapper = passwordWrapper.find("div.bx--form-requirement");
 		expect(messageWrapper).to.have.length(1);
 	});
+	it("Passwordfield eyeIcon tooltip appear correctly", () => {
```
Comment (caritaou) [inline canvas_modules/common-canvas/__tests__/common-properties/controls/passwordfield-test.js:178]: Can you add additional tests?
1. have the application override the labels as a single resource (test using harness)
2. use the defaults when the application didnt specify any

File: canvas_modules/common-canvas/locales/common-properties/locales/en.json [hunk 2]
```diff
@@ -75,6 +75,8 @@
   "expression.maximize.label": "Maximize",
   "toggle.on.label": "On",
   "toggle.off.label": "Off",
+  "password.showPasswordLabel.tooltip":"Show",
+  "password.hidePasswordLabel.tooltip":"Custom Hide Value",
```
Comment (caritaou) [inline canvas_modules/common-canvas/locales/common-properties/locales/en.json:79]: Locales under common-properties are the default if they are not overridden by the application. This should be `Show` and `Hide`

File: canvas_modules/common-canvas/__tests__/common-properties/controls/passwordfield-test.js (hunk 1 above)
Comment (caritaou) [reply canvas_modules/common-canvas/__tests__/common-properties/controls/passwordfield-test.js:178]: You can take a look at this for example: https://github.com/elyra-ai/canvas/blob/main/canvas_modules/harness/src/intl/locales/en.json#L5
I added a custom label in the test harness to simulate an application specifying their own label override for the expression builder. Similarly, applications will specify their own overrides in their resources for the password icon tooltip

**Review by caritaou — CHANGES_REQUESTED** [review]

**Review by tomlyn — CHANGES_REQUESTED** [review]
Hi Nabil 
I noticed the commits for this PR include your commit for changing the toolbar size. This commit should not include commit from previous work items. Can you sort it out so this commit is not included please? Thanks.

image

**Comment by nabilIbm** [comment]
Hi @tomlyn I'm working on creating a new PR

**Comment by nabilIbm** [comment]
CLosed.

**Pull request closed without merging**